src/goose/synopsis/system.py

```diff
@@ -0,0 +1,149 @@
+import json
+from tiktoken import get_encoding
+from exchange import Message
+import subprocess
+import os
+import atexit
+import platform
+from pathlib import Path
+from typing import Dict, Iterable, List, Set
+
+from attrs import define, field
+from exchange.content import ToolUse
+from goose.toolkit.utils import get_language
+
+
+@define
+class File:
+    path: str
+    content: str
+    language: str
+
+    @property
+    def context(self) -> str:
+        return f"""\
+{self.path}
+```{self.language}
+{self.content}
+```"""
+
+
+@define
+class OperatingSystem:
+    """
+    OperatingSystem class that can manage background processes created using subprocess.Popen.
+    """
+
+    cwd: str = os.getcwd()
```

Comment:
:nit: did we forget `field` for these?

Reply:
not sure we need it? i dont think this requires using a factory instead of just a default value, or were you thinking of something else?